ocs_ci/utility/ibmcloud.py

```diff
@@ -520,6 +522,268 @@ def delete_volume_id(self, volume):
+            if instance_name["status"] == constants.STATUS_STOPPED:
+                node_obj = OCP(kind="Node", resource_name=instance_name["name"]).get()
+                node_obj_ocs = OCS(**node_obj)
+                stop_node_list.append(node_obj_ocs)
+            if instance_name["status"] == constants.STATUS_STOPPED:
+                node_obj = OCP(kind="Node", resource_name=instance_name["name"]).get()
+                node_obj_ocs = OCS(**node_obj)
+                stopping_node_list.append(node_obj_ocs)
+                stop_node_list.append(node_obj_ocs)
+        logger.info("Force stopping node which are in stopping state")
+        self.stop_nodes(nodes=stop_node_list, force=True, wait=True)
+        logger.info("Starting Stopped Node")
+        self.start_nodes(nodes=stop_node_list)
+
+    def terminate_nodes(self, nodes, wait=True):
+        """
+        Terminate the Node in IBMCloud
+        Args:
+            nodes (list): The OCS objects of the nodes
+            wait (bool): True in case wait for status is needed, False otherwise
+        """
+        for node in nodes:
+            cmd = f"ibmcloud is instance-delete {node.name} -f"
+            out = run_ibmcloud_cmd(cmd)
+            logger.info(f"Node deletion command output: {out}")
+            break
+
+        if wait:
+            for node in nodes:
+                sample = TimeoutSampler(
+                    timeout=300,
+                    sleep=10,
+                    func=self.check_node_status,
+                    node_name=node.name,
+                )
+                sample.wait_for_func_status(result=True)
+                break
+
+    def detach_volume(self, volume, node=None):
+        """
+        Detach volume from node on IBM Cloud.
+        Args:
+            volume (str): volume id.
+            node (OCS): worker node id to detach.
+        """
+
+        logger.info(f"volume is : {volume}")
+
+        cmd = f"ibmcloud is volume {volume} --output json"
+        out = run_ibmcloud_cmd(cmd)
+        out = json.loads(out)
+
+        if out["status"] == "available":
+            attachment_id = out["volume_attachments"][0]["id"]
+            cmd = (
+                f"ibmcloud is instance-volume-attachment-update {node.name} {attachment_id} "
+                f"--output json --auto-delete=false"
+            )
+            out = run_ibmcloud_cmd(cmd)
+            out = json.loads(out)
+            logger.info(f"Update command output: {out}")
+            cmd = (
+                f"ibmcloud is instance-volume-attachment-detach {node.name} {attachment_id} "
+                f"--output=json --force"
+            )
+            out = run_ibmcloud_cmd(cmd)
+            logger.info(f"detachment command output: {out}")
+
+    def attach_volume(self, volume, node):
+        """
+        Attach volume to node on IBM Cloud.
+        Args:
+            volume (str): volume id.
+            node (OCS): worker node id to attach.
```

Comment:
Typo fix. It should be `worker node object to detach.`

Reply:
i have update the code